pyomo/gdp/basic_step.py

```diff
@@ -57,12 +74,20 @@ def apply_basic_step( disjunctions ):
         #
         ans.disjuncts[idx].src = Block(ans.DISJUNCTIONS)
         for i in ans.DISJUNCTIONS:
-            tmp = _pseudo_clone(disjunctions[i].disjuncts[idx[i]])
+            tmp = _pseudo_clone(disjunctions[i].disjuncts[
+                idx[i] if isinstance(idx, tuple) else idx])
             for k,v in list(iteritems( tmp.component_map() )):
                 if k == 'indicator_var':
                     continue
                 tmp.del_component(k)
                 ans.disjuncts[idx].src[i].add_component(k,v)
+        # Copy in the constraints corresponding to the improper disjunctions
+        ans.disjuncts[idx].improper_constraints = ConstraintList()
+        for constr in constraints:
+            for indx in constr:
+                ans.disjuncts[idx].improper_constraints.add(
+                    (constr[indx].lower, constr[indx].body, constr[indx].upper)
+                )
```

Comment:
Does it make sense to give the user an option to deactivate the original constraint after applying the basic step?

Reply:
Do I not deactivate the original constraint? If so, that's an oversight on my part.